sdk/evaluator.go

```diff
@@ -0,0 +1,146 @@
+// Copyright 2023 Mia srl
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package sdk
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/rond-authz/rond/core"
+	"github.com/rond-authz/rond/openapi"
+	"github.com/rond-authz/rond/types"
+	"github.com/sirupsen/logrus"
+)
+
+type PolicyResult struct {
+	QueryToProxy []byte
+	Allowed      bool
+}
+
+// Warning: This interface is experimental, and it could change with breaking also in rond patches.
+// Do not use outside this repository until it is not ready.
+type Evaluator interface {
+	Config() openapi.RondConfig
+
+	EvaluateRequestPolicy(ctx context.Context, input core.RondInput, userInfo types.User) (PolicyResult, error)
+	EvaluateResponsePolicy(ctx context.Context, input core.RondInput, userInfo types.User, decodedBody any) ([]byte, error)
```

Comment:
It's the Unmarshalled response body. We return []byte since it's the result of the policy. In this way, we have not ownership of the serialization of the body. I better improve the documentation

Reply:
What do you think about the new documentation added to the struct? It is to improve, but it's an initial point